prime-router/src/main/kotlin/Element.kt

```diff
@@ -990,25 +990,26 @@ data class Element(
      * @param allElementValues the values for all other elements which are updated as needed
      * @param schema the schema
      * @param defaultOverrides element name and value pairs of defaults that override schema defaults
-     * @param index the index of the item from a report being processed
+     * @param itemIndex the index of the item from a report being processed
      * @param sender Sender who submitted the data.  Can be null if called at a point in code where its not known
      * @return a mutable set with the processed value or empty string
      */
     fun processValue(
         allElementValues: Map<String, String>,
         schema: Schema,
         defaultOverrides: Map<String, String> = emptyMap(),
-        index: Int = 0,
+        itemIndex: Int,
         sender: Sender? = null,
     ): ElementResult {
+        check(itemIndex > 0) { "Item index was $itemIndex, but must be larger than 0" }
         val retVal = ElementResult(if (allElementValues[name].isNullOrEmpty()) "" else allElementValues[name]!!)
         if (useMapper(retVal.value) && mapperRef != null) {
             // This gets the required value names, then gets the value from mappedRows that has the data
             val args = mapperArgs ?: emptyList()
             val valueNames = mapperRef.valueNames(this, args)
             val valuesForMapper = valueNames.mapNotNull { elementName ->
                 if (elementName.contains("$")) {
-                    tokenizeMapperValue(elementName, index)
+                    tokenizeMapperValue(elementName, itemIndex)
```

Comment:
It’s not just a name change though. It calls check to verify the index is greater than 0. @whytheplatypus is correct. This could potentially cause an issue with what the mappers return.

Reply:
@MauriceReeves-usds So the mappers are called by this function, so this will not affect the mappers.  Only the serializers and report call this function, so this is a way for someone not to make the mistake of changing the index back to 0.